src/VirtoCommerce.DescriptionExportImportModule.Data/Validation/ImportReviewValidator.cs

```diff
@@ -61,6 +67,20 @@ private void AttachValidators()
                         .WithMissingRequiredValueCodeAndMessage("Product SKU")
                         .WithImportState()
             );
+
+            When(x => !string.IsNullOrEmpty(x.Record.ProductSku),
+            () =>
+                RuleFor(x => x.Record.ProductSku)
+                    .MustAsync(async (sku, _) =>
+                    {
+                        var productSearchResult = await _productSearchService.SearchProductsAsync(new ProductSearchCriteria { Take = 0, Skus = new[] { sku }, });
```

Comment:
Move products searching logic to PreValidate of the parent validator. Save the search result to the context. One searching one the batch.